Sources/PerfectLib/JSONConvertible.swift

```diff
@@ -292,6 +292,24 @@ extension Array: JSONConvertible {
     }
 }
 
+extension Set: JSONConvertible {
+    /// Convert a Set into JSON text.
+    public func jsonEncodedString() throws -> String {
+        var s = "["
+        var first = true
+        for v in self {
+            if !first {
+                s.append(",")
```

Comment:
Is there a reason you're using this `for` loop instead of `joined(separator:)`? https://developer.apple.com/documentation/swift/array/1690077-joined

Reply:
Hi Alex, I went with the same (existing) approach used in `Array`'s extension. My pull request was intended to let others know that `Set` is not conforming to protocol `JSONConvertible`, as stated in the JIRA issue I created (ISS-556). Feel free to implement it in any other (better) way. Thanks for looking into it!

Reply:
please adding a test script as well to prove this work. Thank you!